controllers/medusa/cassandrabackup_controller.go

```diff
@@ -75,6 +75,26 @@ func (r *CassandraBackupReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	backup := instance.DeepCopy()
 
+	cassdcKey := types.NamespacedName{Namespace: backup.Namespace, Name: backup.Spec.CassandraDatacenter}
+	cassdc := &cassdcapi.CassandraDatacenter{}
+	err = r.Get(ctx, cassdcKey, cassdc)
+	if err != nil {
+		logger.Error(err, "failed to get cassandradatacenter", "CassandraDatacenter", cassdcKey)
+		return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
+	}
+
+	pods, err := r.getCassandraDatacenterPods(ctx, cassdc, logger)
+	if err != nil {
+		logger.Error(err, "Failed to get datacenter pods")
+		return ctrl.Result{RequeueAfter: r.DefaultDelay}, err
+	}
+
+	// If the CassandraBackup object was created by a sync operation, we mustn't trigger a backup.
```

Comment:
The issue with this approach is that I don't think "backup" definition and "backupJob" should be the same. That would solve this mentioned issue. Also, if metadata per object is necessary, we could use annotation instead of polluting the CRD with extra fields.

Reply:
Right, that change is coming in the next PR 👍